fenix/app/src/test/java/org/mozilla/fenix/shopping/ReviewQualityCheckFeatureTest.kt

```diff
@@ -394,4 +418,138 @@ class ReviewQualityCheckFeatureTest {
         tested.start()
         assertFalse(isExpanded!!)
     }
+
+    @Test
+    fun `GIVEN feature is enabled WHEN non product url accessed THEN callback not called`() {
+        runTest {
+            var invokedCounter = 0
+            val tab = createTab(
+                url = "https://www.mozilla.org",
+                id = "test-tab",
+                isProductUrl = false,
+            )
+            val browserState = BrowserState(
+                tabs = listOf(tab),
+                selectedTabId = tab.id,
+            )
+            val tested = ReviewQualityCheckFeature(
+                appStore = AppStore(),
+                browserStore = BrowserStore(
+                    initialState = browserState,
+                ),
+                shoppingExperienceFeature = FakeShoppingExperienceFeature(),
+                onIconVisibilityChange = {},
+                onBottomSheetStateChange = {},
+                onProductPageDetected = {
+                    invokedCounter++
+                },
+            )
+
+            tested.start()
+
+            assertEquals(invokedCounter, 0)
+        }
+    }
+
+    @Test
+    fun `GIVEN feature is enabled WHEN product url accessed THEN callback called`() {
+        runTest {
+            var invokedCounter = 0
+            val tab = createTab(
+                url = "https://www.shopping.org",
+                id = "test-tab",
+                isProductUrl = true,
+            ).let {
+                it.copy(content = it.content.copy(loading = false, progress = 100))
```

Comment:
We don't have the condition for progress now so do we need this in tests? Same for the other tests below.